server/game/cards/characters/04/qhorinhalfhand.js

```diff
@@ -0,0 +1,37 @@
+const DrawCard = require('../../../drawcard.js');
+
+class QhorinHalfhand extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
+            when: {
+                afterChallenge: (event, challenge) => (
+                    challenge.challengeType === 'military' &&
+                    challenge.winner === this.controller &&
+                    challenge.isParticipating(this)
+                )
+            },
+            handler: () => {
+                this.game.promptForSelect(this.controller, {
+                    activePromptTitle: 'Select a character',
+                    waitingPromptTitle: 'Waiting for opponent to use ' + this.name,
+                    cardCondition: card => (
+                        card.location === 'play area' && 
+                        card.getType() === 'character' && 
+                        !card.isUnique() && 
+                        card.getStrength() < this.getStrength() &&
+                        card.controller !== this.controller),
+                    onSelect: (p, card) => {
+                        this.card.controller.killCharacter(this.card);
```

Comment:
You don't want `this.card` here. You want to kill the card that was selected, which was passed as parameter `card`. Thus, it should be:

```javascript
card.controller.killCharacter(card)
```